tinkerbell/stack/templates/nginx.yaml

```diff
@@ -29,9 +29,7 @@ spec:
         image: {{ .Values.stack.image }}
         command: ["/bin/bash", "-c"]
         args:
-        - export POD_NAMESERVER=$(awk 'NR==2 {print $2}' /etc/resolv.conf);
-          envsubst '$POD_NAMESERVER' < /tmp/nginx.conf.template > /etc/nginx/nginx.conf;
-          nginx -g 'daemon off;'
```

Comment:
The issue that I see with approach is, we are basing our assumptions/configuration based on what the machine has instead of doing it declaratively and really turn the users attention to this, as a potential problem of tink setup if DNS isn't setting correctly. This is really important especially that users don't fully understand how those tink requirements are mainly used and if they overlook it they might end up in a state where things aren't running correctly and there is no real obvious reason why, especially DNS, because 90% of these hidden issue are DNS 😂, the other 10% is certificates 😄. 

This problem will surface even harder when we start using the operator and then we need to handle differently(e.g: setting in CRD for tinkerbell(TBConfiguration).

Reply:
If i understand correctly, you want to be able to specify the DNS server when using`NodeLocalDNS`. Is this not the default case for a Kubernetes cluster. And the current implementation we have to read the nameserver is working well. Making this explicit and manual brings with it the trade off of a higher barrier to entry when deploying the Tinkerbell stack. We have received a lot of feedback about providing sane defaults and making the initial user experience easier. I'm very sensitive to that and agree with Chris that we could improve the command but that it is a very sane default. I dont want to take on that trade off, especially when we can document and show the value that needs to be set if folks have a non default DNS setup.

Reply:
Yeah no worries, I mean it could be something like, if clusterDNS is not set, read the `nameserver` from the `resolve.conf` file. I just wanted to make sure that, such configurations are declaratively chosen, instead of implicitly invoking it. 

> If i understand correctly, you want to be able to specify the DNS server when usingNodeLocalDNS
Not only that, I wanted to let the user choose what is the dns server ip that should be used for the tink stack, instead of implicitly reading that from the `resolve.conf` file
> We have received a lot of feedback about providing sane defaults and making the initial user experience easier.   
Yeah this is a very important point. Alright then, in the template I will check if the helm chart value is present get that if not, simply read from the resolve.conf file.